test/families/location_scale_low_rank.jl

```diff
@@ -0,0 +1,177 @@
+
+@testset "interface LocationScaleLowRank" begin
+    @testset "$(basedist) rank=$(rank) $(realtype)" for basedist in [:gaussian, :gaussian_nonstd],
+        n_rank in [1, 2],
+        realtype in [Float32, Float64]
+
+        n_dims = 10
+        n_montecarlo = 1000_000
+
+        location = randn(realtype, n_dims)
+        scale_diag = ones(realtype, n_dims)
+        scale_factors = randn(realtype, n_dims, n_rank)
+
+        q = if basedist == :gaussian
+            LowRankGaussian(location, scale_diag, scale_factors)
+        elseif basedist == :gaussian_nonstd
+            MvLocationScaleLowRank(
+                location, scale_diag, scale_factors, Normal(realtype(3), realtype(3))
+            )
+        end
+
+        q_true = if basedist == :gaussian
+            μ = location
+            Σ = Diagonal(scale_diag .^ 2) + scale_factors * scale_factors'
+            MvNormal(location, Σ)
+        elseif basedist == :gaussian_nonstd
+            μ = location + scale_diag .* fill(3, n_dims) + scale_factors * fill(3, n_rank)
+            Σ = 3^2 * (Diagonal(scale_diag .^ 2) + scale_factors * scale_factors')
+            MvNormal(μ, Σ)
+        end
+
+        @testset "eltype" begin
+            @test eltype(q) == realtype
+        end
+
+        @testset "logpdf" begin
+            z = rand(q)
+            @test logpdf(q, z) ≈ logpdf(q_true, z) rtol = realtype(1e-2)
+            @test eltype(logpdf(q, z)) == realtype
+
+            @test logpdf(q, z; non_differntiable=true) ≈ logpdf(q_true, z) rtol = realtype(
+                1e-2
+            )
+            @test eltype(logpdf(q, z; non_differntiable=true)) == realtype
+        end
+
+        @testset "entropy" begin
+            @test eltype(entropy(q)) == realtype
+            @test entropy(q) ≈ entropy(q_true)
+        end
+
+        @testset "length" begin
+            @test length(q) == n_dims
+        end
+
+        @testset "statistics" begin
+            @testset "mean" begin
+                @test eltype(mean(q)) == realtype
+                @test mean(q) == mean(q_true)
```

Comment:
```suggestion
                @test mean(q) ≈ mean(q_true)
```
This might fix the failing test.